controller/array_action/array_mediator_svc.py

```diff
@@ -294,6 +297,51 @@ def get_host_by_host_identifiers(self, initiators):
             logger.debug("can not found host by using initiators: {0} ".format(initiators))
             raise controller_errors.HostNotFoundError(initiators)
 
+    def _get_detailed_hosts_list(self):
+        logger.debug("Getting detailed hosts list on array {0}".format(self.endpoint))
+        hosts_list = self.client.svcinfo.lshost()
+        if not hosts_list:
+            return []
+
+        # for each host get its detailed info from array by sending batch of commands
+        detailed_hosts_list_cmd = self._get_detailed_hosts_list_cmd(hosts_list)
+        logger.debug("Sending getting detailed hosts list commands batch")
+        detailed_hosts_list_output, detailed_hosts_list_errors = self.client.send_raw_command(detailed_hosts_list_cmd)
+        detailed_hosts_list_as_string = bytes_to_string(detailed_hosts_list_output)
+        detailed_hosts_list_error_msg = self._get_formatted_hosts_list_error_msg(detailed_hosts_list_errors)
+        if not detailed_hosts_list_errors:
+            logger.error("Errors returned from getting detailed hosts list: {0}".format(detailed_hosts_list_error_msg))
+
+        logger.debug("Reading detailed hosts list commands batch response")
+        hosts_reader = SVCListResultsReader(detailed_hosts_list_as_string)
+        res = []
+        for host_details in hosts_reader:
+            host_id = host_details.get(HOST_ID_PARAM)
+            host_name = host_details.get(HOST_NAME_PARAM)
+            iscsi_names = host_details.get_as_list(HOST_ISCSI_NAMES_PARAM)
+            wwns = host_details.get_as_list(HOST_WWPNS_PARAM)
+            host = Host(host_id, host_name, iscsi_names, wwns)
+            res.append(host)
+        return res
+
+    def _get_detailed_hosts_list_cmd(self, host_list):
+        writer = StringIO()
+        for host in host_list:
+            host_id = host.get(HOST_ID_PARAM)
+            writer.write(LIST_HOSTS_CMD)
+            writer.write(" ")
+            writer.write(host_id)
+            writer.write(LIST_CMDS_SEPARATOR)
+        return writer.getvalue()
+
+    def _get_formatted_hosts_list_error_msg(self, detailed_host_list_errors_bytes):
+        detailed_host_list_errors = bytes_to_string(detailed_host_list_errors_bytes)
+        if not detailed_host_list_errors:
+            return ""
+        elif len(detailed_host_list_errors) <= MAX_HOSTS_LIST_ERR_MSG_LENGTH:
```

Comment:
this can be shortened to:
`return textwrap.shorten(errors, width=MAX, placeholder="...")`

Reply:
seem s it doesn't work
>>textwrap.shorten('abcdefgh', width=5, placeholder="...")
'...'

Reply:
my bad.
try `'{}...'.format(textwrap.wrap('abcdefgh', width=5)[0], '...')`

Reply:
This is the same thing i did except it is not one line
i pefere m y way

Reply:
I agree that the logic is very similar, but what about the naming?
could you make the names more general? e.g.:
`_get_formatted_hosts_list_error_msg` -> `truncate`
`detailed_host_list_errors` -> `error_msgs`

Reply:
I agree the logic is similar.
but could you still rename it to something more general? e.g.:
`_get_formatted_hosts_list_error_msg` -> `truncate_errors`
`detailed_host_list_errors` -> `errors_text`

btw I don't see this function being called anywhere.

Reply:
done

Reply:
@igorgonibm why don't you use `:` slicing?